In [20]:
import pandas as pd
import numpy as np

# I) PREPARING & CLEANING

## I.1) Grouping the different sheets that ar part of the workbook

In [10]:
# Ruta del archivo Excel de entrada
archivo_entrada = './data/raw data.xlsx'

# Cargar todas las hojas en un diccionario
hojas = pd.read_excel(archivo_entrada, sheet_name=None)

# Unir todas las hojas en un solo DataFrame
df_completo = pd.concat(hojas.values(), ignore_index=True)

# Guardar el DataFrame combinado en un nuevo archivo Excel
archivo_salida = './data/raw data mixed.xlsx'
df_completo.to_excel(archivo_salida, index=False)

print(f"Las hojas fueron combinadas exitosamente en {archivo_salida}")

Las hojas fueron combinadas exitosamente en ./data/raw data mixed.xlsx


## I.1) Manual edition: due errors by users, correction was need once every sheet is in a one workbook

In [86]:
path = './data/prepared data mixed.xlsx' 
df = pd.read_excel(path)
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3053 entries, 0 to 3052
Data columns (total 7 columns):
 #   Column                             Non-Null Count  Dtype 
---  ------                             --------------  ----- 
 0   Numero OT                          3053 non-null   object
 1   Fecha recepcion                    3053 non-null   object
 2   Cliente                            3051 non-null   object
 3   Tipo de ingreso                    3039 non-null   object
 4   Responsable de ingreso             3040 non-null   object
 5   Descripcion del proyecto           3048 non-null   object
 6   Dimensiones [mm] anchoxlargoxalto  1384 non-null   object
dtypes: object(7)
memory usage: 167.1+ KB


In [87]:
#df = df.iloc[:,:7]
#df = df.dropna(axis = 1, how = 'all')
df = df.sort_values(by = 'Fecha recepcion')
df = df.reset_index(drop=True)
df

Numero OT Fecha recepcion            Cliente Tipo de ingreso  \
0         5579      2021-03-02        Frau Albors           Nuevo   
1         5580      2021-03-02           All Pack           Nuevo   
2         5581      2021-03-02   Olivos del valle             NaN   
3      5587-40      2021-03-03           New mont           Nuevo   
4         5591      2021-03-03            Newmont           Nuevo   
...        ...             ...                ...             ...   
3048      8328      2025-01-27  TECSESI (techint)           Nuevo   
3049      8327      2025-01-27  TECSESI (techint)           Nuevo   
3050      8331      2025-01-27  TECSESI (techint)           Nuevo   
3051      8330      2025-01-27  TECSESI (techint)           Nuevo   
3052      8329      2025-01-27  TECSESI (techint)           Nuevo   

     Responsable de ingreso            Descripcion del proyecto  \
0          Gustavo Martinez                 Cilindro hidráulico   
1          Gustavo Martinez                        Kit de sello   
2          Gustavo Martinez                             Con coe   
3                       NaN        cil extension de manipulador   
4          Gustavo Martinez              Cilindro tolva sandvik   
...                     ...                                 ...   
3048      Guillermo Sanchez          martillo hidráulico 771045   
3049      Guillermo Sanchez        martillo hidráulico E&W RX18   
3050      Guillermo Sanchez       criquet hidroneumático ph30-c   
3051      Guillermo Sanchez     criquet hidroneumático torletti   
3052      Guillermo Sanchez  martillo cat H75E mas componentes    

     Dimensiones [mm] anchoxlargoxalto  
0                                  NaN  
1                                  NaN  
2                                  NaN  
3                                  NaN  
4                         2000x300x300  
...                                ...  
3048                      350x1400x800  
3049                      600x3300x750  
3050                               NaN  
3051                               NaN  
3052                               NaN  

[3053 rows x 7 columns]

## I.3) EXPLORATION

- ### **a) How many clients the company have?**

In [88]:
df['Cliente'].unique().shape[0]

580

### But some of the names are duplicated by misswritting. So we will get how many categories we really have:

In [93]:
df['Cliente'] = df['Cliente'].str.upper()
df['Cliente'].value_counts()

Cliente
MINA PIRQUITAS             233
YMAD                       154
FINNING                     94
HOLCIM                      85
L'EQUIPE MONTEUR            81
                          ... 
VIMECO                       1
INGLOBAL                     1
HIDROTEC PERFORACIONES       1
LA MARQUESINA                1
MINERA SAN PEDRO             1
Name: count, Length: 488, dtype: int64

### We found:
- ### *There is 488 different clients*
- ### *Most common customer: Mina Pirquitas (~16% of all the projects)**

(*) Because, so far, the company count 3053 projects. Counting since the beginning of 2021.


- ### **b) How many projects by year & month take the company?**

In [ ]:
df['año y mes'] = df['Fecha recepcion'].str.split('-', expand = True)[0].value_counts()

0
2023    849
2022    840
2021    708
2024    621
2025     35
Name: count, dtype: int64

In [111]:
df['año y mes'] = df['Fecha recepcion'].str.split('-', expand = True)[0] + '-' + df['Fecha recepcion'].str.split('-', expand = True)[1]
df.head(4)

Numero OT Fecha recepcion           Cliente Tipo de ingreso  \
0      5579      2021-03-02       FRAU ALBORS           Nuevo   
1      5580      2021-03-02          ALL PACK           Nuevo   
2      5581      2021-03-02  OLIVOS DEL VALLE             NaN   
3   5587-40      2021-03-03          NEW MONT           Nuevo   

  Responsable de ingreso      Descripcion del proyecto  \
0       Gustavo Martinez           Cilindro hidráulico   
1       Gustavo Martinez                  Kit de sello   
2       Gustavo Martinez                       Con coe   
3                    NaN  cil extension de manipulador   

  Dimensiones [mm] anchoxlargoxalto año y mes  
0                               NaN   2021-03  
1                               NaN   2021-03  
2                               NaN   2021-03  
3                               NaN   2021-03

In [121]:
test = df['año y mes'].value_counts()#sort_values()
test = pd.DataFrame(test)
test

count
año y mes       
2021-09      106
2023-05      102
2023-07       96
2022-03       91
2021-08       90
2021-10       88
2023-06       86
2022-11       86
2022-01       82
2022-09       80
2021-03       80
2022-04       74
2022-05       74
2022-07       74
2023-08       74
2023-10       73
2023-03       73
2021-12       72
2022-06       71
2023-12       71
2024-04       69
2024-10       67
2023-04       66
2022-10       66
2021-06       65
2023-02       63
2023-01       62
2024-03       60
2024-01       60
2021-07       59
2022-12       58
2021-11       56
2023-09       54
2024-09       54
2024-06       54
2022-08       50
2024-12       49
2021-04       49
2024-11       46
2024-02       45
2024-05       43
2021-05       43
2024-07       42
2025-01       35
2022-02       34
2024-08       32
2023-11       29